clang/lib/Frontend/InitPreprocessor.cpp

```diff
@@ -467,6 +467,7 @@ static void InitializeStandardPredefinedMacros(const TargetInfo &TI,
       Builder.defineMacro("CL_SYCL_LANGUAGE_VERSION", "121");
       Builder.defineMacro("SYCL_LANGUAGE_VERSION", "201707");
     } else if (LangOpts.SYCLVersion == 2020) {
+      Builder.defineMacro("CL_SYCL_LANGUAGE_VERSION", "2020");
```

Comment:
I'm not sure if this is by-spec.

Reply:
SYCL 1.2.1 defines `CL_SYCL_LANGUAGE_VERSION`, whereas SYCL 2020 provisional defines `SYCL_LANGUAGE_VERSION`. I'm not sure that this is correct.

Reply:
The change looks ok to me but I don't know what the value for CL_SYCL_LANGUAGE_VERSION should be. @Fznamznon would you know?

Reply:
Oops. I see you commented right before I did.

> SYCL 1.2.1 defines `CL_SYCL_LANGUAGE_VERSION`, whereas SYCL 2020 provisional defines `SYCL_LANGUAGE_VERSION`. I'm not sure that this is correct.

SYCL 1.2.1  defines `CL_SYCL_LANGUAGE_VERSION` and `SYCL_LANGUAGE_VERSION`.